libs/dex/src/main/java/io/crate/data/SentinelRow.java

```diff
@@ -39,7 +39,7 @@ public Object get(int index) {
 
         @Override
         public Object[] materialize() {
-            throw new IllegalStateException("Iterator is not on a row");
+            return new Object[0];
         }
```

Comment:
What's the purpose of this change? 

I wonder if not reflecting this change in numColumns/get can lead to sonme issues because of mixed semantics?

Reply:
The BatchIteratorTester now uses .map to go from Row -> Object[] via materialize and it asserts that `currentElement` can be called in some off-row scenarios or after close. 

> I wonder if not reflecting this change in numColumns/get can lead to sonme issues because of mixed semantics?

Should be fine as the SentinelRow is only used if in an off position.